urwid/widget/widget_decoration.py

```diff
@@ -30,6 +30,17 @@ class WidgetDecoration(Widget):  # "decorator" was already taken
     """
 
     def __init__(self, original_widget: Widget) -> None:
+        # TODO(Aleksei): reduce amount of multiple inheritance usage
+        # Special case: subclasses with multiple inheritance causes `super` call wrong way
```

Comment:
What's an example of this? `super()` MRO should resolve `__init__` call order for diamond inheritance cases

Reply:
For `LineBox` it's resolved incorrectly.
Technically `LineBox` using circular inheritance from `Widget`, but in this case `super` call in `WidgetDecoration` really calls like from `LineBox` to the `WidgetWrap`:

```
  File "/home/alekseis/PycharmProjects/urwid/urwid/widget/line_box.py", line 124, in __init__
    WidgetDecoration.__init__(self, original_widget)
  File "/home/alekseis/PycharmProjects/urwid/urwid/widget/widget_decoration.py", line 37, in __init__
    super().__init__()
TypeError: WidgetWrap.__init__() missing 1 required positional argument: 'w'
```